File: 2020/Day01.py
def part_1(input):
    return next(i*(2020-i) for i in input if 2020-i in input)

def part_2(input):
    return next(i*j*(2020-i-j) for i in input for j in input if 2020-i-j in input)

with open("2020/Day01_input.txt") as f:
    input = set(map(int, f.read().split()))

    print(f"Part 1: {part_1(input)}")
    print(f"Part 2: {part_2(input)}")

File: 2022/Day01.go
package main

import (
	"fmt"
	"os"
	"slices"
	"strconv"
	"strings"
)

func totals(in string) []int {
	var t []int
	for _, b := range strings.Split(in, "\n\n") {
		s := 0
		for _, l := range strings.Fields(b) {
			n, _ := strconv.Atoi(l)
			s += n
		}
		t = append(t, s)
	}
	slices.Sort(t)
	return t
}

func part1(in string) int {
	t := totals(in)
	return t[len(t)-1]
}

func part2(in string) int {
	t := totals(in)
	return t[len(t)-1] + t[len(t)-2] + t[len(t)-3]
}

func main() {
	b, _ := os.ReadFile("2022/Day01_input.txt")
	in := strings.TrimRight(string(b), "\n")
	fmt.Println("Part 1:", part1(in))
	fmt.Println("Part 2:", part2(in))
}
